test/1_test_fullpath.js

```diff
@@ -124,23 +152,30 @@ contract('Voucher tests', (addresses) => {
 
   describe('Direct minting', function () {
     it('must fail: unauthorized minting ERC-1155', async () => {
-      await truffleAssert.reverts(
-        contractERC1155ERC721.mint(users.attacker.address, 666, 1, []),
-        truffleAssert.ErrorType.REVERT
-      );
+      await expect(
+        contractERC1155ERC721.functions[fnSignatures.mint1155](
```

Comment:
Why is the call being done using the function signature instead of just calling the function. This Waffle matcher can be called in the following way:

```
await expect(token.transfer(walletTo.address, 1007))
  .to.be.revertedWith('Insufficient funds');
```

The function can be called directly.

Reply:
`transfer` might be called directly, but identical functions with different signatures could not, hence we need to get their signature explicitly